src/java.base/share/classes/java/net/Inet4Address.java

```diff
@@ -134,6 +160,55 @@ class Inet4Address extends InetAddress {
         holder().originalHostName = hostName;
     }
 
+    /**
+     * Creates an {@code Inet4Address} based on the provided {@linkplain
+     * Inet4Address##format textual representations} of an IPv4 address.
+     * <p> If the provided IPv4 address literal cannot represent a {@linkplain
+     * Inet4Address##format valid IPv4 address} an {@code IllegalArgumentException} is thrown.
+     * <p> This method doesn't block, i.e. no reverse lookup is performed.
+     *
+     * @param ipv4AddressLiteral the textual representation of an IPv4 address.
+     * @return an {@link Inet4Address} object with no hostname set, and constructed
+     *         from the provided IPv4 address literal.
+     * @throws IllegalArgumentException if the {@code ipv4AddressLiteral} cannot be
+     *         parsed as an IPv4 address literal.
+     * @throws NullPointerException if the {@code ipv4AddressLiteral} is {@code null}.
+     */
+    public static Inet4Address ofLiteral(String ipv4AddressLiteral) {
+        Objects.requireNonNull(ipv4AddressLiteral);
+        return parseAddressString(ipv4AddressLiteral, true);
+    }
+
+    /**
+     * Parses the given string as an IPv4 address literal.
+     * If the given {@code addressLiteral} string cannot be parsed as an IPv4 address literal
+     * and {@code throwIAE} is {@code false}, {@code null} is returned.
+     * If the given {@code addressLiteral} string cannot be parsed as an IPv4 address literal
+     * and {@code throwIAE} is {@code true}, an {@code IllegalArgumentException} is thrown.
+     * If string contains an {@linkplain IPAddressUtil#validateNumericFormatV4(String, boolean)
+     * ambiguous literal} - {@code IllegalArgumentException} is thrown irrelevant to
```

Comment:
```suggestion
     * Otherwise, if it can be considered as {@linkplain IPAddressUtil#validateNumericFormatV4(String, boolean)
     * an ambiguous literal} - {@code IllegalArgumentException} is thrown irrelevant to
```